pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -298,12 +298,52 @@ type SecuritySpec struct {
 	// +optional
 	// +nullable
 	KeyManagementService KeyManagementServiceSpec `json:"kms,omitempty"`
-	// KeyRotation defines options for Key Rotation.
+	// KeyRotation defines options for rotation of OSD disk encryption keys.
 	// +optional
 	// +nullable
 	KeyRotation KeyRotationSpec `json:"keyRotation,omitempty"`
 }
 
+// ClusterSecuritySpec is the CephCluster security spec to include various security items such as kms
+type ClusterSecuritySpec struct {
+	SecuritySpec `json:""`
+
+	// CephX configures CephX key settings. More: https://docs.ceph.com/en/latest/dev/cephx/
+	// +optional
+	CephX ClusterCephxConfig `json:"cephx,omitempty"`
+}
+
+type ClusterCephxConfig struct {
+	// Daemon configures CephX key settings for local Ceph daemons managed by Rook and part of the
+	// Ceph cluster. Daemon daemon CephX keys can be rotated without affecting client connections.
+	Daemon CephxConfig `json:"local,omitempty"`
+}
+
+type CephxConfig struct {
+	// KeyRotationPolicy controls if and when CephX keys are rotated after initial creation.
+	// One of Disabled, WithCephVersionUpdate, or KeyGeneration. Default Disabled.
+	// +optional
+	// +kubebuilder:validation:Enum="";Disabled;WithCephVersionUpdate;KeyGeneration
+	KeyRotationPolicy CephxKeyRotationPolicy `json:"keyRotationPolicy,omitempty"`
+
+	// KeyGeneration specifies the desired CephX key generation. This is used when KeyRotationPolicy
```

Comment:
TODO: SEL rule to prevent users from decreasing

Reply:
@travisn what are your thoughts on a rule to enforce +1 incrementing of this value?

I think preventing decreasing makes sense, but I'm not sure about limiting this to +1 only.

Reply:
I do like the idea of the rule to help make it incremental. That said, it may not work if the rotation policy is changing. For example:
- Start with a rotation policy of KeyGeneration, with the generation set to 1
- Change the policy to update with ceph upgrades. After one or more ceph upgrades, the generation will be incremented.
- Change the policy back to KeyGeneration. But the generation cannot be incremented directly to the new expected value.

Reply:
Yeah, I think that's a good identification of the challenging corner case. For now, I will stick to just preventing decreases. This could be refined later if we have time and desire.